### PR TITLE
HDDS-5607. remove container manager v1 code

### PR DESCRIPTION
**Pull request by JacksonYao287**
## What changes were proposed in this pull request?

remove container manager v1 code and rename containerManagerV2 to containerManager

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-5607

## How was this patch tested?

unit test

### CODE REVIEW

**Comment by JacksonYao287** [comment]
@bshashikant PTAL!

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/RatisPipelineUtils.java [hunk 1]
```diff
@@ -99,24 +99,6 @@ static void destroyPipeline(DatanodeDetails dn, PipelineID pipelineID,
...
-  /**
-   * Return the list of pipelines who share the same set of datanodes
-   * with the input pipeline.
-   *
-   * @param stateManager PipelineStateManager
-   * @param pipeline input pipeline
-   * @return list of matched pipeline
-   */
-  static List<Pipeline> checkPipelineContainSameDatanodes(
-      PipelineStateManager stateManager, Pipeline pipeline) {
```
Comment (bharatviswa504) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/RatisPipelineUtils.java:111]: Is this not required?

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/hdds/scm/container/TestContainerStateManagerIntegration.java [hunk 2]
```diff
@@ -102,7 +100,7 @@ public void testAllocateContainer() throws IOException {
             SCMTestUtils.getReplicationFactor(conf), OzoneConsts.OZONE);
     ContainerInfo info = containerManager
         .getMatchingContainer(OzoneConsts.GB * 3, OzoneConsts.OZONE,
-            container1.getPipeline());
+            container1.getPipeline(), excludedContainerIDS);
```
Comment (bharatviswa504) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/hdds/scm/container/TestContainerStateManagerIntegration.java:103]: We still have older method with out needing to pass excludedContainerIDS right?

**Comment by bharatviswa504** [comment]
Thank You @JacksonYao287 overall LGTM, few minor comments/questions

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/RatisPipelineUtils.java (hunk 1 above)
Comment (JacksonYao287) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/RatisPipelineUtils.java:111]: yes, it seems not used. for now , we use the following one instead
```
  static List<Pipeline> checkPipelineContainSameDatanodes(
      StateManager stateManager, Pipeline pipeline) {
```

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/hdds/scm/container/TestContainerStateManagerIntegration.java (hunk 2 above)
Comment (JacksonYao287) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/hdds/scm/container/TestContainerStateManagerIntegration.java:103]:
```
//V1 code 
  ContainerInfo getMatchingContainer(long size, String owner,
      Pipeline pipeline);

//V2 code
  ContainerInfo getMatchingContainer(long size, String owner,
                                     Pipeline pipeline,
                                     Set<ContainerID> excludedContainerIDS);
```

v1 code is not used any more, so after we remove v1 code, we could used V2 method instead.

Comment (bharatviswa504) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/hdds/scm/container/TestContainerStateManagerIntegration.java:103]: We still have old method which is defined in interface ContainerManager.java

  default ContainerInfo getMatchingContainer(long size, String owner,
                                     Pipeline pipeline) {
    return getMatchingContainer(size, owner, pipeline, Collections.emptySet());
  }

But I think that is fine, as anyway for tests we are passing empty hashset for excludedContainerIDS

**Review by bharatviswa504 — APPROVED** [review]
+1 LGTM

**Comment by bharatviswa504** [comment]
Thank You @JacksonYao287 for the contribution.

**Pull request merged**